recipes/fuse_mnt.rb

```diff
@@ -0,0 +1,160 @@
+include_recipe "hops::default"
+
+# Intall the needed packages
+case node['platform_family']
+when 'debian'
+  package 'fuse'
+  package 'libfuse-dev'
+when 'rhel'
+  package 'fuse'
+  package 'fuse-libs'
+end
+
+# Allow non root users to mount file systems 
+bash "fuseconfig" do
+  user "root" 
+  code <<-EOH
+    echo "user_allow_other" >>  /etc/fuse.conf
+  EOH
+  not_if "grep '^user_allow_other' /etc/fuse.conf"
+end
+
+directory node['hops']['fuse']['staging_folder'] do
+  action :create
+  owner node['hops']['hdfs']['user']
+  group node['hops']['group']
+  mode "0750"
+  recursive true
+end
+
+directory node['hops']['fuse']['mount_point'] do
+  action :create
+  owner node['hops']['hdfs']['user']
+  group node['hops']['group']
+  mode "0700"
+  recursive true
+end
+
+# download fuse mount bin
+fuse_mount_bin_url = node['hops']['fuse']['dist_url']
+bin_name = File.basename(fuse_mount_bin_url)
+fuse_mount_bin = "#{node['hops']['sbin_dir']}/#{bin_name}"
+
+file fuse_mount_bin do
+  action :delete
+  only_if { File.exist? "#{fuse_mount_bin}" }
+end
+
+remote_file fuse_mount_bin do
+  source fuse_mount_bin_url
+  owner node['hops']['hdfs']['user']
+  group node['hops']['group']
+  mode '0755'
+  action :create
+end
+
+# FS URI
+my_ip = my_private_ip()
+if node['install']['localhost'].casecmp?("true")
+  nn_address = "localhost"
+else 
+  if node['hops']['nn']['private_ips'].include?(my_ip)
+    # If I'm a NameNode set it to my fqdn or IP
+    if service_discovery_enabled()
+      nn_address = node['fqdn']
+    else
+      nn_address = my_ip
+    end
+  else
+    # Otherwise use Service Discovery FQDN
+    nn_address = rpc_namenode_fqdn
+  end
+end
+
+# creating script to mount FS
+file "#{node['hops']['sbin_dir']}/mount-fs.sh" do
+  action :delete
+  only_if { File.exist? "#{node['hops']['sbin_dir']}/mount-fs.sh" }
+end
+
+file "#{node['hops']['sbin_dir']}/unmount-fs.sh" do
+  action :delete
+  only_if { File.exist? "#{node['hops']['sbin_dir']}/unmount-fs.sh" }
+end
```

Comment:
Why do you delete them?

Reply:
Because I did not remember that chef replaces the file if it exists. I have removed it